src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1639 @@
+    SET_STOLEN_ITEM_2(STOLEN_ITEM_NONE);
+
+    // ??
+    func_801149A0(ITEM_OCARINA_FAIRY, SLOT_TRADE_DEED);
+    func_801149A0(ITEM_SLINGSHOT, SLOT_TRADE_KEY_MAMA);
+    func_801149A0(ITEM_LONGSHOT, SLOT_TRADE_COUPLE);
+
+    for (j = 1; j < 4; j++) { // s1
+        if (GET_CUR_FORM_BTN_ITEM(j) >= ITEM_MOON_TEAR && GET_CUR_FORM_BTN_ITEM(j) <= ITEM_PENDANT_MEMORIES) {
+            SET_CUR_FORM_BTN_ITEM(j, ITEM_NONE);
+            func_80112B40(globalCtx, j);
+        }
+    }
+
+    gSaveContext.save.skullTokenCount &= ~0xFFFF0000;
+    gSaveContext.save.skullTokenCount &= ~0x0000FFFF;
+    gSaveContext.save.unk_EC4 = 0;
+
+    gSaveContext.save.unk_E88[0] = 0;
+    gSaveContext.save.unk_E88[1] = 0;
+    gSaveContext.save.unk_E88[2] = 0;
+    gSaveContext.save.unk_E88[3] = 0;
+    gSaveContext.save.unk_E88[4] = 0;
+    gSaveContext.save.unk_E88[5] = 0;
+    gSaveContext.save.unk_E88[6] = 0;
+
+    func_80143A54();
+
+    for (i = 0; i < 8; i++) {
+        gSaveContext.save.inventory.dungeonItems[i] &= (u8)~1;
+        gSaveContext.save.inventory.dungeonKeys[i] = 0;
+        gSaveContext.save.inventory.strayFairies[i] = 0;
+    }
+
+    gSaveContext.save.playerData.rupees = 0;
+    gSaveContext.save.unk_F65 = 0;
+    gSaveContext.powderKegTimer = 0;
+    gSaveContext.unk_1014 = 0;
+    gSaveContext.jinxTimer = 0;
+    gSaveContext.rupeeAccumulator = 0;
+
+    func_800F3B2C(globalCtx);
+}
+
+void Sram_IncrementDay(void) {
+    if (CURRENT_DAY <= 3) {
+        gSaveContext.save.day++;
+        gSaveContext.save.daysElapsed++;
+    }
+
+    gSaveContext.save.bombersCaughtNum = 0;
+    gSaveContext.save.bombersCaughtOrder[0] = 0;
+    gSaveContext.save.bombersCaughtOrder[1] = 0;
+    gSaveContext.save.bombersCaughtOrder[2] = 0;
+    gSaveContext.save.bombersCaughtOrder[3] = 0;
+    gSaveContext.save.bombersCaughtOrder[4] = 0;
+    // Unconfirmed: "Bombers Hide & Seek started on Day 1???"
+    gSaveContext.save.weekEventReg[73] &= (u8)~0x10;
+    // Unconfirmed: "Bombers Hide & Seek in Progress"
+    gSaveContext.save.weekEventReg[85] &= (u8)~0x02;
+}
+
+u16 Sram_CalcChecksum(void* data, size_t count) {
+    u8* dataPtr = data;
+    u16 chkSum = 0;
+
+    while (count-- > 0) {
+        chkSum += *dataPtr;
+        dataPtr++;
+    }
+    return chkSum;
+}
+
+// Resets `Save` substruct
+void Sram_ResetSave(void) {
+    gSaveContext.save.entranceIndex = 0x1C00;
+    gSaveContext.save.equippedMask = 0;
+    gSaveContext.save.isFirstCycle = false;
+    gSaveContext.save.unk_06 = 0;
+    gSaveContext.save.linkAge = 0;
+    gSaveContext.save.isNight = false;
+    gSaveContext.save.daySpeed = 0;
+    gSaveContext.save.snowheadCleared = 0;
+    gSaveContext.save.hasTatl = false;
+    gSaveContext.save.isOwlSave = false;
+
+    // Instead of bloating all save context accesses with an extra sub-struct, the size of the would-be sub-struct
+    // is calculated manually
+    bzero(&gSaveContext.save.playerData, sizeof(Save) - offsetof(Save, playerData));
```

Comment:
I'm fine with this solution, but curious if OOT's solution of temporary structs in `z_sram.c` was considered.

Reply:
Yeah, but I don't really like it, because it introduces new structs which we would need to keep in sync with the main SaveContext struct.

Reply:
Ah, didn't think of that. Okay, this solution seems better :) .